controller/controller-http/src/main/java/com/pinterest/rocksplicator/controller/resource/Clusters.java

```diff
@@ -510,8 +510,27 @@ public Response registerHost(@PathParam("namespace") String namespace,
   @Path("/hosts/{namespace: [a-zA-Z0-9\\-_]+}/{clusterName: [a-zA-Z0-9\\-_]+}")
   @Produces(MediaType.APPLICATION_JSON)
   public Response getHosts(@PathParam("namespace") String namespace,
-                           @PathParam("clusterName") String clusterName) {
-    Set<HostBean> hosts = clusterManager.getHosts(new Cluster(namespace, clusterName), false);
+                           @PathParam("clusterName") String clusterName,
+                           @QueryParam("excludeBlacklisted") Optional<Boolean> excludeBlacklisted,
+                           @QueryParam("excludeHostsInConfig") Optional<Boolean> excludeHostsInConfig) {
+    Set<HostBean> hosts = clusterManager.getHosts(new Cluster(namespace, clusterName), excludeBlacklisted.orElse(false));
+    if (excludeHostsInConfig.isPresent() && excludeHostsInConfig.get().equals(true)) {
+      final ClusterBean clusterBeanInConfig;
+      try {
+        clusterBeanInConfig = checkExistenceAndGetClusterBean(namespace, clusterName);
+      } catch (Exception e) {
+        String message = String.format("Failed to read from zookeeper: %s", e);
+        LOG.error(message);
+        return Utils.buildResponse(HttpStatus.INTERNAL_SERVER_ERROR_500,
+                ImmutableMap.of("message", message));
+      }
+      if (!hosts.removeAll(clusterBeanInConfig.getHosts())) {
```

Comment:
if cannot, we should not make it an error, it just mean there is no idle hosts. return an empty set.